src/jit/codegenarm64.cpp

```diff
@@ -2616,9 +2616,15 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                 if (targetType == TYP_STRUCT)
```

Comment:
I think the check here has to be TYP_STRUCT.  SIMD type structs are typed as SIMD8/12/16/32 and loaded with a single instruction typically from memory.